modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala

```diff
@@ -47,9 +47,9 @@ final case class Execution(execution: List[Action]) {
     * If the index doesn't exist, `Current` is returned unmodified.
     */
   def mark(i: Int)(r: Result): Execution =
-    Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = actionStateFromResult(r)), execution).getOrElse(execution))
+    Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = actionStateFromResult(r)(a.state)), execution).getOrElse(execution))
 
-  def start(i: Int): Execution = Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = Action.Started), execution).getOrElse(execution))
+  def start(i: Int): Execution = Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = a.state.copy(runState = Action.Started)), execution).getOrElse(execution))
```

Comment:
Is this a `Monocle` lens?

Reply:
I don't know. I just copied the code from `mark`.